src/mgmt_conn.c

```diff
@@ -1588,6 +1588,16 @@ process_message(uzfs_mgmt_conn_t *conn)
 			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED, hdrp);
 			break;
 		}
+		if (resize_data->size < ZVOL_VOLUME_SIZE(zinfo->main_zv)) {
+			LOGERRCONN(conn, "Failed to resize main volume %s, "
+			    "resizing from %lu to %lu is not allowed",
+			    zinfo->main_zv->zv_name,
```

Comment:
why don't you just do zinfo->name?
and, put this log after drop_refcnt

Reply:
good to have drop_refcnt after logging since we are using `ZVOL_VOLUME_SIZE(zinfo->main_zv)` to get the size.

Reply:
> why don't you just do zinfo->name?
Not aware of that I will update it

Reply:
Done both the changes